ratis-server/src/main/java/org/apache/ratis/server/raftlog/RaftLogBase.java

```diff
@@ -394,6 +396,9 @@ public LogEntryProto getEntry(TimeDuration timeout) throws RaftLogIOException, T
       } catch (TimeoutException t) {
         final String err = getName() + ": Timeout readStateMachineData for " + toLogEntryString(logEntry);
         LOG.error(err, t);
```

Comment:
I forgot to mention last time -- let's move LOG.error  inside the if-statement, i.e.
```
          if (timeout.compareTo(readStatemachineTimeout) > 0) {
            getRaftLogMetrics().onStateMachineReadTimeout();
            final String err = getName() + ": Timeout readStateMachineData for " + toLogEntryString(logEntry);
            LOG.error(err, t);
         }
```

Reply:
Sure.   LOG in LogAppenderBase is alread refined by removing the "follower.getName()" in LOG, for "follower.getName() is already displayed in the daemon thread name. See following LOG sample. 

2021-12-07 19:58:23,035 [98e5b27a-c3e9-4f86-ab85-b2caf84f012b@group-34584CA620E5->5a4a8be1-c921-4ca7-af7c-62a37a55cab7-GrpcLogAppender-LogAppenderDaemon] ERROR org.apache.ratis.server.raftlog.RaftLog: 98e5b27a-c3e9-4f86-ab85-b2caf84f012b@group-34584CA620E5-SegmentedRaftLog: Timeout readStateMachineData for (t:1, i:251185), STATEMACHINELOGENTRY, cmdType: WriteChunk traceID: "" containerID: 14793 datanodeUuid: "98e5b27a-c3e9-4f86-ab85-b2caf84f012b" pipelineID: "347d76ee-be5d-4880-998a-34584ca620e5" writeChunk { blockID { containerID: 14793 localID: 6383 } chunkData { chunkName: "chunk-28" offset: 28311552 len: 1048576 checksumData { type: CRC32 bytesPerChecksum: 1048576 checksums: "\236\224o\322" } } }, container path=/data9/hdds/hdds/CID-d78e4370-93d0-42ad-89b5-d71a6a798a73/current/containerDir28/14793
java.util.concurrent.TimeoutException
        at java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1771)
        at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1915)
        at org.apache.ratis.server.raftlog.RaftLogBase$EntryWithDataImpl.getEntry(RaftLogBase.java:395)
        at org.apache.ratis.util.DataQueue.pollList(DataQueue.java:137)
        at org.apache.ratis.server.leader.LogAppenderBase.newAppendEntriesRequest(LogAppenderBase.java:150)
        at org.apache.ratis.grpc.server.GrpcLogAppender.appendLog(GrpcLogAppender.java:213)
        at org.apache.ratis.grpc.server.GrpcLogAppender.run(GrpcLogAppender.java:147)
        at org.apache.ratis.server.leader.LogAppenderDaemon.run(LogAppenderDaemon.java:77)
        at java.lang.Thread.run(Thread.java:748)

Reply:
Sure, thanks,